cpp/src/arrow/CMakeLists.txt

```diff
@@ -490,6 +490,10 @@ if(ARROW_BUILD_STATIC AND WIN32)
   target_compile_definitions(arrow_static PUBLIC ARROW_STATIC)
 endif()
 
+if(NOT ARROW_UTF8PROC_USE_SHARED AND WIN32)
+  target_compile_definitions(arrow_static PRIVATE UTF8PROC_STATIC)
```

Comment:
Could you try `INTERFACE_COMPILER_DEFINITIONS`?
https://github.com/apache/arrow/pull/7818#discussion_r459082935

Reply:
@kou In `Findutc8proc.cmake`? I'll see if I can get that to work

Reply:
Yes.